### ISSUE TITLE
Doesn't integrate with Neovim

### ISSUE THREAD

**Issue by stevendobay** [ensime/ensime-vim#340]
Yes, I've seen #94 but it doesn't help.

Setup:

``` sh
sudo add-apt-repository -y ppa:neovim-ppa/unstable
sudo apt-get update
sudo apt-get install neovim vim-nox-py2 python-pip python-dev python-software-properties
pip2 install websocket-client sexpdata neovim
```

init.vim ->

``` vim
call plug#begin('~/.config/nvim/plugged')
Plug 'bling/vim-airline'
Plug 'ensime/ensime-vim'
Plug 'derekwyatt/vim-scala' , {'for' : 'scala'}
call plug#end()
```

call 

``` vim
:w
:source %
:PlugInstall
:UpdateRemotePlugins " tried to copy the rplugin directory before
```

ensimeConfig and
- [x] works with vim-nox-py2
- [ ] works with neovim(:EnInstall is an unknown command but :h ensime gives the docs)

Ubuntu 16.04, neovim 0.16, +python, -python3

Results: 3 times(2 diff. machines) EnInstall was an unknown command and the 4th:

```
Error detected while processing CursorHold Auto commands for "*.scala":
error caught while executing async callback:
TypeError("'str' object is not callable",)
TypeError: 'str' object is not callable

the call was requested at
  File ".../.config/nvim/plugged/ensime-vim/ensime_shared/util.py", line 51, in catch
    handler(str(e))
  File ".../.config/nvim/plugged/ensime-vim/ensime_shared/client.py", line 270, in disable_completely
    self.disable_plugin()
  File "..../.config/nvim/plugged/ensime-vim/ensime_shared/client.py", line 242, in disable_plugin
    threadsafe_vim('set runtimepath-={}'.format(path))
  File ".../.config/nvim/plugged/ensime-vim/ensime_shared/client.py", line 238, in threadsafe_vim
    self.vim.session.threadsafe_call(command)
```

**Comment by ches** [comment]
I've tried to replicate your setup with an Ubuntu 16.04 VM (except that I don't use plug.vim), and I haven't been able to reproduce this.

You must restart nvim after running `:UpdateRemotePlugins`, you probably know this but it's not precisely clear from your steps, so just mentioning. I don't know what your comment `" tried to copy the rplugin directory before` means, but it sounds like something unnecessary and perhaps problematic.

What does your rplugin manifest generated by the command look like? On my setup this is at `~/.local/share/nvim/rplugin.vim`.

If you can reproduce getting as far as the backtrace you posted, can see and share more of that trace?

**Comment by stevendobay** [comment]
Hi @ches !
 I've restarted it - I forgot to mention, I thought it's clear because it was after the ":source %" command.
The rplugin copy was the manual fix for #94 - but it doesn't work for me.
I've created a new setup:

UpdateRemotePlugins:

````
remote/host: python host registered plugins ['ensime.py']
remote/host: generated rplugin manifest: /home/steven/.local/share/nvim/rplugin.vim
```

EnInstall:

```
Error detected while processing function remote#define#CommandBootstrap:
line    5:
no request handler registered for "/home/steven/.config/nvim/plugged/ensime-vim/rplugin/python/ensime.py:command:EnInstall"
Encountered ImportError loading plugin at /home/steven/.config/nvim/plugged/ensime-vim/rplugin/python/ensime.py: No module named ensime
Traceback (most recent call last):
  File "/home/steven/.local/lib/python2.7/site-packages/neovim/plugin/host.py", line 131, in _load
    file, pathname, descr = find_module(name, [directory])
ImportError: No module named ensime
```

rplugin manifest:

```
" python3 plugins


" ruby plugins


" python plugins
call remote#host#RegisterPlugin('python', '/home/steven/.config/nvim/plugged/ensime-vim/rplugin/python/ensime.py', [
      \ {'sync': v:true, 'name': 'BufLeave', 'type': 'autocmd', 'opts': {'pattern': '*.scala', 'eval': 'expand("<afile>")'}},
      \ {'sync': v:true, 'name': 'CursorHold', 'type': 'autocmd', 'opts': {'pattern': '*.scala', 'eval': 'expand("<afile>")'}},
      \ {'sync': v:true, 'name': 'CursorMoved', 'type': 'autocmd', 'opts': {'pattern': '*.scala', 'eval': 'expand("<afile>")'}},
      \ {'sync': v:true, 'name': 'VimEnter', 'type': 'autocmd', 'opts': {'pattern': '*.scala', 'eval': 'expand("<afile>")'}},
      \ {'sync': v:true, 'name': 'VimLeave', 'type': 'autocmd', 'opts': {'pattern': '*.scala', 'eval': 'expand("<afile>")'}},
      \ {'sync': v:true, 'name': 'EnAddImport', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnClients', 'type': 'command', 'opts': {'nargs': '0', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDebugBacktrace', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDebugClearBreaks', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDebugContinue', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDebugNext', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDebugSetBreak', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDebugStart', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDebugStep', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDebugStepOut', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDeclaration', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDeclarationSplit', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDocBrowse', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnDocUri', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnFormatSource', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnInline', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnInspectType', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnInstall', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnOrganizeImports', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnShowPackage', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnRename', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnSuggestImport', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnSearch', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnSymbol', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnSymbolByName', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnToggleFullType', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnToggleTeardown', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnType', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnTypeCheck', 'type': 'command', 'opts': {'nargs': '*', 'range': ''}},
      \ {'sync': v:true, 'name': 'EnCompleteFunc', 'type': 'function', 'opts': {}},
      \ {'sync': v:true, 'name': 'EnPackageDecl', 'type': 'function', 'opts': {}},
     \ ])
```

**Comment by stevendobay** [comment]
Hi @ches!
 I've reinstalled it a hundred times and I've succeeded on my home computer. I've tried to debug it before but I couldn't find anything and it still doesn't work on the same platform on a virtualbox instance - I think the installation can fail at many points without notifying the user. Sometime the help is missing for ensime or it isn't registered in the rplugin manifest or it says the rplugin generation failed. Sometime it fails on code completion. I don't like to read python code but I'll let you know if I find anything.
